### ISSUE TITLE
seq2seq/sequence_to_sequence_implementation.ipynb  throws error

### ISSUE THREAD

**Issue by joe-of-all-trades** [udacity/deep-learning#163]
I ran into this error while running the script. I didn't modify anything. 
![image](https://user-images.githubusercontent.com/21017820/27814628-bb765210-60b8-11e7-9c9e-b5e19a76d317.png)

How should this problem be solved ?

**Comment by ShreyashSoni** [comment]
Facing the same error. I think it is because of the TensorFlow version.

**Comment by NawfalTachfine** [comment]
The seq2seq lesson and translation project both require TF 1.1.0

**Comment by NawfalTachfine** [comment]
Whereas previous notebooks need TF 1.0.0
It's not explicitly mentioned anywhere. That's the only way I managed to get them to work.

**Comment by ShreyashSoni** [comment]
The latest TF version that I installed is 1.2.1
It seems to throw that error.

**Comment by NawfalTachfine** [comment]
Yup. It doesn't work with the latest. you need to specify the version as 1.1.0
`pip3 install tensorflow-gpu==1.1.0`

**Comment by ShreyashSoni** [comment]
anything we can do to keep up with all the updates of TF?

**Comment by NawfalTachfine** [comment]
It's to do with the way the notebooks were written in the first place. TF is moving quickly and updates are not always backward compatible. It's challenging enough to actually solve the exercises and debug one's own code. I prefer the simple route of replicating the environment the code was first written in instead of modifying it to work with the latest TF.
Lessons and projects are usually independent so using different environments for each doesn't have any bad repercussions.

**Comment by joe-of-all-trades** [comment]
changing the codes to the following solves the problem for me 

> training_decoder_output = tf.contrib.seq2seq.dynamic_decode(training_decoder, impute_finished=True,
                                                                       maximum_iterations=max_target_sequence_length)
training_decoder_output = training_decoder_output[0]
inference_decoder_output = tf.contrib.seq2seq.dynamic_decode(inference_decoder,
                                                            impute_finished=True, minimum_iterations=max_target_sequence_length)
inference_decoder_output = inference_decoder_output[0]

However, I couldn't tell if the codes were working properly... With the input 'hello', it outputs 'ehllo'  Training loss did go down, though.

**Comment by NawfalTachfine** [comment]
looks good @joe-of-all-trades, nice work!

**Comment by mcleonard** [comment]
Hi all! Fixed this in 1fd809014f24640b74a7dff6bfaddbcb8665b393
